packages/server/prisma/schema.prisma

```diff
@@ -12,15 +12,18 @@ datasource db {
 }
 
 model User {
-  id          String    @default(uuid()) @id
+  id          String @unique @default(uuid())
   project_id  String
-  username    String?
-  email       String?
+  username    String? @unique
+  email       String? @unique
   password    String
-  role        Int       @default(0)
+  role        Int?      @default(0)
   created_at  DateTime  @default(now())
   updated_at  DateTime  @default(now())
   deleted_at  DateTime?
   reset_code            String?
   reset_code_expires_at DateTime?
+
+  @@id([id, project_id])
+  @@unique([id, project_id, email, username])
```

Comment:
Maybe using id alone as primary key will be enough. Uniqueness of the combination does not seems necessary, since we are doing the uniqueness check in UserService.

Reply:
Fixed.